lib/curl_fnmatch.c

```diff
@@ -140,33 +141,33 @@ static char_class charclass(unsigned char c)
 }
 
 /* Include a character or a range in set. */
-static void setcharorrange(unsigned char **pp, unsigned char *charset)
+static void setcharorrange(char **pp, char *charset)
 {
-  unsigned char *p = (*pp)++;
-  unsigned char c = *p++;
+  char *p = (*pp)++;
+  char c = *p++;
 
-  charset[c] = 1;
+  charset[(unsigned int)c] = 1;
   if(ISALNUM(c) && *p++ == '-') {
     char_class cc = charclass(c);
-    unsigned char endrange = *p++;
+    char endrange = *p++;
 
     if(endrange == '\\')
       endrange = *p++;
     if(endrange >= c && charclass(endrange) == cc) {
       while(c++ != endrange)
         if(charclass(c) == cc)  /* Chars in class may be not consecutive. */
-          charset[c] = 1;
+          charset[(unsigned int)c] = 1;
```

Comment:
I still don't see the point in this change. It still *adds* two typecasts, that's contrary to the main motivation for this PR.

Reply:
It has to be because char is signed